torch/csrc/autograd/variable.h

```diff
@@ -175,6 +175,9 @@ inline Variable make_variable(at::Tensor data, VarFlags flags=DEFAULT_FLAGS,
     // don't expose 0-dim tensors to Variable API.
     data = data.as_strided_({1}, {1});
   }
+  if (!flags.requires_grad) {
+    grad_fn = nullptr;
+  }
```

Comment:
Should we assert that it's never used with a grad_fn, when you don't have this flag set? It seems like the function was created unnecessarily.